chia/full_node/mempool.py

```diff
@@ -328,38 +324,60 @@ def at_full_capacity(self, cost: int) -> bool:
 
         return self.total_mempool_cost() + cost > self.mempool_info.max_size_in_cost
 
+    def get_items_spending_coin_ids(self, spent_coin_ids: List[bytes32]) -> List[MempoolItem]:
+        items: List[MempoolItem] = []
+        for coin_ids in chunks(spent_coin_ids, SQLITE_MAX_VARIABLE_NUMBER):
+            args = ",".join(["?"] * len(coin_ids))
+            with self._db_conn:
+                cursor = self._db_conn.execute(
+                    f"SELECT * FROM tx WHERE name IN (SELECT tx FROM spends WHERE coin_id IN ({args}))", tuple(coin_ids)
+                )
+                items.extend(self._row_to_item(row) for row in cursor)
+        return items
+
     def create_bundle_from_mempool_items(
         self, item_inclusion_filter: Callable[[bytes32], bool]
-    ) -> Optional[Tuple[SpendBundle, List[Coin], List[Coin]]]:
+    ) -> Optional[Tuple[SpendBundle, List[Coin]]]:
         cost_sum = 0  # Checks that total cost does not exceed block maximum
         fee_sum = 0  # Checks that total fees don't exceed 64 bits
-        spend_bundles: List[SpendBundle] = []
-        removals: List[Coin] = []
+        processed_spend_bundles = 0
         additions: List[Coin] = []
-        log.info(f"Starting to make block, max cost: {self.mempool_info.max_block_clvm_cost}")
-        for item in self.spends_by_feerate():
-            if not item_inclusion_filter(item.name):
+        # This contains a map of coin ID to a coin spend solution and its isolated cost
+        eligible_coin_spends = EligibleCoinSpends()
+        coin_spends: List[CoinSpend] = []
+        sigs: List[G2Element] = []
+        with self._db_conn:
+            cursor = self._db_conn.execute("SELECT name, fee FROM tx ORDER BY fee_per_cost DESC, seq ASC")
+        for row in cursor:
```

Comment:
it looks a bit funky to have this loop outside of the context manager. I'm sure it's correct, but I'm also thinking that when we just read the DB, maybe we don't even need the context manager